pkg/cloudprovider/provider/hetzner/provider.go

```diff
@@ -176,6 +182,34 @@ func (p *provider) Create(machine *v1alpha1.Machine, userdata string) (instance.
 		return nil, hzErrorToTerminalError(err, "failed to get server type")
 	}
 
+	tmpRSAKeyPair, err := rsa.GenerateKey(rand.Reader, privateRSAKeyBitSize)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create private RSA key: %v", err)
+	}
+
+	if err := tmpRSAKeyPair.Validate(); err != nil {
+		return nil, fmt.Errorf("failed to validate private RSA key: %v", err)
+	}
+	pubKey, err := ssh.NewPublicKey(&tmpRSAKeyPair.PublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate ssh public key: %v", err)
+	}
+
+	hkey, res, err := client.SSHKey.Create(ctx, hcloud.SSHKeyCreateOpts{
+		Name:      fmt.Sprintf("machine-controller-temp-%s", machine.Spec.Name),
```

Comment:
not sure if this is enough, I can give the same name to a machine in my cluster.